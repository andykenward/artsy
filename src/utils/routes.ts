import queryString from 'query-string';
import { isNil, pick, pickBy } from 'ramda';

const allowedFilters = ['color'];

interface Filter {
  color?: string;
}

export const validateFilterObject = (
  searchQuery: string | undefined
): Filter | null => {
  if (!searchQuery || searchQuery === '') return null;
  const filter = pick(allowedFilters, queryString.parse(searchQuery));
  return pickBy((v) => !isNil(v), filter);
};

export const getFilter = (
  searchQuery: string | undefined
): { color?: string } | null => {
  return validateFilterObject(searchQuery);
};
