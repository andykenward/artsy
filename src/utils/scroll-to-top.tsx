import { RouteComponentProps } from '@reach/router';
import React, { useEffect } from 'react';

export const ScrollToTop: React.FC<RouteComponentProps> = ({
  children,
  location,
}) => {
  const { pathname } = location!;
  useEffect(() => {
    window.scrollTo(0, 0);
  }, [pathname]);

  return <>{children}</>;
};
