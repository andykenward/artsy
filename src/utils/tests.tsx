import { injectGlobalStyles, Theme } from '@artsy/palette';
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from '@reach/router';
import { render } from '@testing-library/react';
import React from 'react';

export const { GlobalStyles } = injectGlobalStyles(`
  @import url("https://webfonts.artsy.net/all-webfonts.css")
`);

export const renderWithTheme = (element: React.ReactElement<any>) =>
  render(
    <Theme>
      <>
        <GlobalStyles />
        {element}
      </>
    </Theme>
  );

export const renderWithRouter = (
  element: React.ReactElement<any>,
  {
    route = '/',
    history = createHistory(createMemorySource((route = '/'))),
  } = {}
) => ({
  ...render(<LocationProvider history={history}>{element}</LocationProvider>),
  // adding `history` to the returned utilities to allow us
  // to reference it in our tests (just try to avoid using
  // this to test implementation details).
  history,
});
