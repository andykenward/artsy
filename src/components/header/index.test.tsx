import React from 'react';
import { Header } from '.';
import { renderWithTheme } from '../../utils/tests';

describe('<Header />', () => {
  it('renders correctly', () => {
    const { container } = renderWithTheme(<Header />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
