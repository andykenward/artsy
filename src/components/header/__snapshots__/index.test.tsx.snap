// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
.c0 {
  margin: 1rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 2rem;
  line-height: 2.375rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin: 0;
}

.c1 {
  color: #000;
  -webkit-transition: color 0.25s;
  transition: color 0.25s;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #000;
}

@media screen and (min-width:768px) {
  .c2 {
    font-size: 2.75rem;
  }
}

@media screen and (min-width:768px) {
  .c2 {
    line-height: 3.125rem;
  }
}

<header
  class="Flex-cw39ct-0 c0"
  display="flex"
>
  <a
    class="c1"
    href="https://artsy.net"
    rel="noopener noreferrer"
    target="_blank"
  >
    <h1
      class="c2 Typography-sc-1pie2ao-1"
      display="inline-flex"
      font-family="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
      font-size="2rem,2.75rem"
    >
      Artsy.net
    </h1>
  </a>
</header>
`;
